great_expectations/core/batch_manager.py

```
@@ -0,0 +1,164 @@
+            batch_list = []
+
+        self._batch_cache = OrderedDict()
+        self._batch_data_cache = {}
+
+        self.load_batch_list(batch_list=batch_list)
+        if len(batch_list) > 1:
+            logger.info(
+                f"{len(batch_list)} batches will be added to this Validator.  The batch_identifiers for the active "
+                f"batch are {self.active_batch.batch_definition['batch_identifiers'].items()}"
+            )
+
+    @property
+    def batch_data_cache(self) -> Dict[str, BatchData]:
+        """Dictionary of loaded BatchData objects."""
+        return self._batch_data_cache
+
+    @property
+    def loaded_batch_ids(self) -> List[str]:
+        """IDs of loaded BatchData objects."""
+        return list(self._batch_data_cache.keys())
+
+    @property
+    def active_batch_data_id(self) -> Optional[str]:
+        """The batch id for the default batch data.
+
+        When a specific Batch is unavailable, then the data associated with the active_batch_data_id will be used.
+        """
+        if self._active_batch_data_id is not None:
+            return self._active_batch_data_id
+
+        if len(self._batch_data_cache) == 1:
+            return list(self._batch_data_cache.keys())[0]
+
+        return None
+
+    @property
+    def active_batch_data(self) -> Optional[BatchData]:
+        """The BatchData object from the currently-active Batch object."""
+        if self.active_batch_data_id is None:
+            return None
+
+        return self._batch_data_cache.get(self.active_batch_data_id)
+
+    @property
+    def batch_cache(self) -> Dict[str, Batch]:
+        """Getter for ordered dictionary (cache) of "Batch" objects in use (with batch_id as key)."""
+        return self._batch_cache
+
+    @property
+    def active_batch_id(self) -> Optional[str]:
+        """Getter for active Batch ID"""
+        active_batch_data_id: str = self._active_batch_data_id
+        if active_batch_data_id != self._active_batch_id:
```

Do we ever expect `self._active_batch_data_id` and `self._active_batch_id` to differ? This makes it seem like they shouldn't except maybe locally in some method (and if so we could use a local for one of these). Basically I'm curious:
1. Should these ever differ?
2. If they differ is this an error and should we die?
3. Do we need 2 variables?

---

@billdirks I believe that I had written up an answer in another place in this PR -- in addition, your question is important: yes, `active_batch_data_id` and `active_batch_id` can be different.  The former refers to the most recent loaded `BatchData`, while the latter refers to the most recent requested `Batch`.  There can be situations where no new `BatchData` objects have been loaded; however, a new `Validator` was instantiated and requested `Batch` objects, whose `BatchData` has already been loaded (and cached).  Since `BatchData` IDs are from the same name space as `Batch` IDs, this helps avoid unnecessary loading of data from the backends.  Thanks.

---

Why do we log a warning? This sounds like a normal situation. Should we log at all? If we want to log, can we do it on the info level?

---

@alexsherstinsky says this only happens in tests. We should file a ticket to clean this up in tests and then we can throw an error here.